tests/run_tests.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# make test
+
+set -u
+set -eE
+
+[[ -z $(which bats) ]] && echo "bats not found, install: https://github.com/bats-core/bats-core#installation" && exit 1
```

Comment:
```
# Test if a command exists.
try_command() {
    command -v "$1" >/dev/null 2>&1
}
```
This is a preferred way of checking if commands exist. It's built into shells vs an external binary. 
https://stackoverflow.com/questions/37056192/which-vs-command-v-in-bash

Reply:
Done